File: trunk/net.tenteCsharp/src_code/smartEnergyControl/HouseGateway.java


cclass smartEnergyControl.SmartEnergyControl; 

public cclass HouseGateway extends TypeComponent{ 

	public SmartPort smartEnergyPort;
		public SmartNotifyPort smartEnergyNotifyPort;
	
	
	public HouseGateway (String id){
		super(id);
	}
	
	public SmartPort getSmartEnergyPort(){
		return smartEnergyPort;
	}
	
	public void setSmartEnergyPort(HouseGateway.SmartPort value){
		this.smartEnergyPort=value;
	}
	
	public SmartNotifyPort getSmartEnergyNotifyPort(){
		return smartEnergyNotifyPort;
	}
	
	public void setSmartEnergyNotifyPort(HouseGateway.SmartNotifyPort value){
		this.smartEnergyNotifyPort=value;
	}
	

	
	
	public cclass Room {
		public boolean smartControlActive;

		public Room (){
				
		}
		public boolean getSmartControlActive(){
			return smartControlActive;
		}
	
		public void setSmartControlActive(boolean value){
			this.smartControlActive=value;
		}
	



	}
	
	public Room getRoomInstance(){
		Room room=new Room();
		return room;
	}	
	
	public cclass Floor {
		public boolean smartControlActive;

		public Floor (){
				
		}
		public boolean getSmartControlActive(){
			return smartControlActive;
		}
	
		public void setSmartControlActive(boolean value){
			this.smartControlActive=value;
		}
	



	}
	
	public Floor getFloorInstance(){
		Floor floor=new Floor();
		return floor;
	}	
	
	public cclass House {
		public boolean smartControlActive;

		public House (){
				
		}
		public boolean getSmartControlActive(){
			return smartControlActive;
		}
	
		public void setSmartControlActive(boolean value){
			this.smartControlActive=value;
		}
	



	}
	
	public House getHouseInstance(){
		House house=new House();
		return house;
	}	
	
	public cclass SmartPort extends TypePort implements ISmartEnergy{


		public SmartPort (){
			super();

		}


		public void activateSmartControl(String floorId,String roomId){
		
		}

		public void deactivateSmartControl(String floorId,String roomId){
		
		}

	}	
	
	public cclass SmartNotifyPort extends TypePort{

		public ArrayList portsISmartEnergyNotify;

		public SmartNotifyPort (){
			super();
			portsISmartEnergyNotify=new ArrayList();

		}


			
		public void connectPort(ISmartEnergyNotify port){
			portsISmartEnergyNotify.add(port);
		}
	}

}

File: pseudoCodeGeneratorElio/src-gen/smartEnergyControl/HouseGateway.cs
using System;
using System.Collections;
using System.Text;

namespace SmartHome
{
	public partial class HouseGateway : TypeComponent
	{
		public SmartPort smartEnergyPort;
			public SmartNotifyPort smartEnergyNotifyPort;
	
		public HouseGateway (String id) 
			: base(id)
		{
		}
		public SmartPort getSmartEnergyPort()
		{
			return smartEnergyPort;
		}
	
		public void setSmartEnergyPort(HouseGateway.SmartPort value)
		{
			this.smartEnergyPort=value;
		}
	
		public SmartNotifyPort getSmartEnergyNotifyPort()
		{
			return smartEnergyNotifyPort;
		}
	
		public void setSmartEnergyNotifyPort(HouseGateway.SmartNotifyPort value)
		{
			this.smartEnergyNotifyPort=value;
		}
	



		public class Room_smartEnergyControl 
		{		
			public Boolean smartControlActive; 

			public Room_smartEnergyControl()
			{
			}
			public Boolean   getSmartControlActive()
			{
				return smartControlActive;
			}
			public void setSmartControlActive(Boolean value)
			{
				this.smartControlActive=value;
			}
	

		}
			
			public Room getRoomInstance()
			{
				Room room=new Room();
				return room;
			}

		public class Floor_smartEnergyControl 
		{		
			public Boolean smartControlActive; 

			public Floor_smartEnergyControl()
			{
			}
			public Boolean   getSmartControlActive()
			{
				return smartControlActive;
			}
			public void setSmartControlActive(Boolean value)
			{
				this.smartControlActive=value;
			}
	

		}
			
			public Floor getFloorInstance()
			{
				Floor floor=new Floor();
				return floor;
			}

		public class House_smartEnergyControl 
		{		
			public Boolean smartControlActive; 

			public House_smartEnergyControl()
			{
			}
			public Boolean   getSmartControlActive()
			{
				return smartControlActive;
			}
			public void setSmartControlActive(Boolean value)
			{
				this.smartControlActive=value;
			}
	

		}
			
			public House getHouseInstance()
			{
				House house=new House();
				return house;
			}	
		
		public class SmartPort : TypePort , ISmartEnergy
		{
 
			public SmartPort()
				: base()
			{

			}


		public void activateSmartControl(String floorId,String roomId)
			{
		
			}

		public void deactivateSmartControl(String floorId,String roomId)
			{
		
			}
	
		}	
		
		public class SmartNotifyPort : TypePort
		{
 		public ArrayList portsISmartEnergyNotify = new ArrayList();

			public SmartNotifyPort()
				: base()
			{
				portsISmartEnergyNotify=new ArrayList();

			}


			public ArrayList getPortsISmartEnergyNotify()
			{
	        	return portsISmartEnergyNotify;
	        }
				
			public void connectPort(ISmartEnergyNotify port)
			{
				portsISmartEnergyNotify.Add(port);
			}
	
		}
	}
}